core/block_validator.go

```diff
@@ -17,31 +17,57 @@
 package core
 
 import (
+	"context"
 	"fmt"
+	"time"
 
+	"github.com/Onther-Tech/plasma-evm/accounts/abi/bind"
 	"github.com/Onther-Tech/plasma-evm/consensus"
+	"github.com/Onther-Tech/plasma-evm/contracts/plasma/rootchain"
 	"github.com/Onther-Tech/plasma-evm/core/state"
 	"github.com/Onther-Tech/plasma-evm/core/types"
+	"github.com/Onther-Tech/plasma-evm/ethclient"
+	"github.com/Onther-Tech/plasma-evm/log"
 	"github.com/Onther-Tech/plasma-evm/params"
 )
 
+var baseCallOpt = &bind.CallOpts{Pending: false, Context: context.Background()}
+
 // BlockValidator is responsible for validating block headers, uncles and
 // processed state.
 //
 // BlockValidator implements Validator.
 type BlockValidator struct {
-	config *params.ChainConfig // Chain configuration options
-	bc     *BlockChain         // Canonical block chain
-	engine consensus.Engine    // Consensus engine used for validating
+	config    *params.ChainConfig // Chain configuration options
+	bc        *BlockChain         // Canonical block chain
+	engine    consensus.Engine    // Consensus engine used for validating
+	rootchain *rootchain.RootChain
 }
 
 // NewBlockValidator returns a new block validator which is safe for re-use
 func NewBlockValidator(config *params.ChainConfig, blockchain *BlockChain, engine consensus.Engine) *BlockValidator {
+	// Dial rootchain provider
+	rootchainBackend, err := ethclient.Dial(config.RootchainURL)
+	if err != nil {
+		log.Error("failed to make rootchainBackend", "err", err)
+		return nil
+	}
+	log.Info("Rootchain provider connected", "url", config.RootchainURL)
+
+	// Instantiate RootChain contract
+	rootchainContract, err := rootchain.NewRootChain(config.RootchainAddress, rootchainBackend)
+	if err != nil {
+		log.Error("failed to make rootchainContract", "err", err)
+		return nil
+	}
+
```

Comment:
If `rootchain` is needed only for validator, just add it as argument of `newBlockValidator` (as well as `NewBlockChain`).

It makes us keep `params.ChainConfig` untouched.